app/models/user_block.rb

```diff
@@ -27,6 +27,12 @@ def active?
   end
 
   ##
+  # returns true if the block is a "zero hour" block
+  def zero_hour?
+    needs_view && (ends_at.to_i == updated_at.to_i)
```

Comment:
If I understand things correctly we have roughly three cases that we care about

1. needs_view set, duration of block ~0 -> "zero hour block"
2. needs_view set, duration of block >> 0 -> longish block were we still want to communicate to the user
3. needs_view not set -> no messaging to the user

Currently all three cases get the same message. With this PR (1) gets a slightly friendlier message than (2) and (3), but we could further differentiate these if we wanted to ie remove the reference to having to log in.